py/dml/ctree.py

```diff
@@ -1602,10 +1608,18 @@ def make(cls, site, lh, rh):
                                                  AddressOfMethod))
                     return mkBoolConstant(site, False)
         if lhtype.is_int:
+            if rh.constant and rhtype.is_arith and (
+                    not (lhtype.min() <= rh.value <= lhtype.max())
+                    or (rhtype.is_float and rh.value % 1)):
+                report(WTYPELIMITS(site, rh, lhtype))
             lh_maybe_negative = lhtype.signed
             lh = as_int(lh)
             lhtype = realtype(lh.ctype())
         if rhtype.is_int:
+            if lh.constant and lhtype.is_arith and (
+                    not (rhtype.min() <= lh.value <= rhtype.max())
+                    or (lhtype.is_float and lh.value % 1)):
```

Comment:
Huh. `x % 1` does actually seem to Just Work, even for nan and inf. I'm surprised.
My earlier way to go about this was `not math.isfinite(x) or x != int(x)`. Your idiom has the benefit of `x` only occurring once, at the cost of being more obtuse.

Reply:
I think I'm fine with the `% 1` route.

Reply:
Good point, inf/nan deserve test cases.

Reply:
Indeed broken, I get no warning from the following:
```
    local int64 b = -1;
    assert b < inf;
    assert inf > b;
    assert b > -inf;
    assert b != -inf;
    assert !(b < nan);
    assert !(b > nan);
    assert b != nan;
```